README.md

```diff
@@ -17,102 +16,112 @@ gem 'nxt_http_client'
-    )
+    config.json_request = true
+    config.json_response = true
+    config.bearer_auth = ENV['USER_SERVICE_API_TOKEN']
     config.x_request_id_proc = -> { ('a'..'z').to_a.shuffle.take(10).join }
   end
 
+  # You may add a log handler if you wish...
   log do |info|
     Rails.logger.info(info)
   end
 
+  # ...as well as a response handler
   response_handler do |handler|
     handler.on(:error) do |response|
-      Raven.extra_context(error_details: error.to_h)
+      Sentry.set_extras(error_details: error.to_h)
       raise StandardError, "I can't handle this: #{response.code}"
     end
   end
 end
 ```
 
-### HTTP Methods
-
-In order to build a request and execute it NxtHttpClient implements all http standard methods.
+and then child classes for accessing specific endpoints and adding custom behaviours.
 
 ```ruby
-class Client < NxtHttpClient
-  def initialize(url)
-    @url = url
-  end
-
-  attr_reader :url
-
-  def fetch
-    get(url) do
-      handler.on(:success) { |response| response.body }
-    end
+class UserFetcher < UserServiceClient
+  def initialize(id)
+    @url = ".../users/#{id}"
   end
 
-  def create(params)
-    post(url, params: params) do
-      handler.on(:success) { |response| response.body }
+  def fetch_email
+    get(url, { fields: :email }) do |response_handler|
+      response_handler.on(:success) do |response|
+        JSON(response.body)['email']
+      end
     end
   end
 
-  def update(params)
-    put(url, params: params) do
-      handler.on(:success) { |response| response.body }
+  def fetch_user_details
+    get(url) do |response_handler|
+      response_handler.on(:success) do |response|
+        body = JSON(response.body)
+        User.new(body)
+      end
     end
   end
 
-  # ... there are others as you know ...
+  private attr_reader :url
 end
 ```
 
+Usage:
+
+```ruby
+client = UserFetcher.new('1234')
+client.fetch_email
+client.fetch_user_details
+```
+
 ### configure
 
-Register your default request options on the class level. Available options are `request_options` that are passed
-directly to the underlying Typhoeus Request. Then there is `base_url` and `x_request_id_proc`.
+Register your default request options on the class level. Available options are:
+- `request_options`, passed directly to the underlying Typhoeus Request
+- `base_url=`
+- `x_request_id_proc=`
+- `json_request=`: Shorthand to set the Content-Type request header to JSON and automatically convert request bodies to JSON
+- `json_response=`: Shorthand to set the Accept request header and automatically convert success response bodies to JSON
+- `raise_response_errors=`: Makes the client raise a `NxtHttpClient::Error` for a non-success response. 
+  You can also do this manually by setting a response_handler.
+- `bearer_auth=`: Set a bearer token to be sent in the Authorization header
+- `basic_auth=`: Pass an array containing username and password, to be sent as Basic credentials in the Authorization header
```

Comment:
I'd prefer (1) `config.basic_auth = { username:, password: }` or (2) `config.basic_auth = [username, password]` over (3) `config.basic_auth = username, password`.
For one, I find the 3 a bit confusing for the same reason I find a method call with multiple arguments without parenthesis confusing (and this technically is one). And the (1) makes it crystal clear which one is which, although, it should be usually clear without keywords.

Reply:
Sure, I can switch to (1).

Reply:
Changed to a Hash: https://github.com/nxt-insurance/nxt_http_client/pull/158/commits/895317d0ed7582ca1a6bf0219b4e3a4293e7eb1c